### ISSUE TITLE
ReloadData() will cause additional expanded view

### ISSUE THREAD

**Issue by kohdesmond** [younatics/YNExpandableCell#5]
sorry to give u bad news again.

If expanded view is open, .reloadData() will cause it to remain and present.

Tapping on the expandableCell will open another detail page

![screen shot 2017-03-21 at 16 39 37](https://cloud.githubusercontent.com/assets/1192544/24138904/03466d52-0e55-11e7-953e-7f55032db724.png)

**Comment by younatics** [comment]
Got it! No problem. I will tell you again if it is fixed. :)

**Comment by younatics** [comment]
Anyway, if you developed that cool feature(Down Arrow), could you introduce it for me? Thanks 👍

**Comment by kohdesmond** [comment]
sure, will share you the code

**Comment by kohdesmond** [comment]
@younatics 

here's the rotate code

```
func rotateAnimation(_ img: UIImageView, duration: Double, isClose: Bool) {
        
        let angle = isClose == true ? CGFloat(M_PI) : CGFloat(-2 * M_PI)
        
        UIView.animate(withDuration: duration, animations: {
            img.transform = CGAffineTransform(rotationAngle: angle)
        })
    }
    
    func tableView(_ tableView: YNTableView, didDeselectRowAt indexPath: IndexPath, isExpandedCell: Bool)
    {
        if !isExpandedCell
        {
            self.taskTableView.deselectRow(at: indexPath, animated: true)
            print("selected no expandable cell")
        }
        else
        {
            guard let selectedCell = taskTableView.cellForRow(at: indexPath) as? ExpandableTaskCell else { return }
            isClose = selectedCell.upArrow.isHidden == true ?  true : false
            
            if !isClose
            {  print("close Deselect")
                selectedCell.upArrow.isHidden = true
                self.rotateAnimation(selectedCell.downArrow, duration: 0.3, isClose: false)
            }
        }
    }
    
    func tableView(_ tableView: YNTableView, didSelectRowAt indexPath: IndexPath, isExpandedCell: Bool)
    {
        if !isExpandedCell
        {
            self.taskTableView.deselectRow(at: indexPath, animated: true)
            print("selected no expandable cell")
        }
        else
        {
            guard let selectedCell = taskTableView.cellForRow(at: indexPath) as? ExpandableTaskCell else { return }
           isClose = selectedCell.upArrow.isHidden == true ?  true : false
            if isClose
            {   print("open Select")
                selectedCell.upArrow.isHidden = false
                self.rotateAnimation(selectedCell.downArrow, duration: 0.3, isClose: true)
            }
        }
    }
```

**Comment by younatics** [comment]
I override reload data method. It will work :)
Please see  [v0.6.0](https://github.com/younatics/YNExpandableCell/releases/tag/0.6.0)

**Comment by kohdesmond** [comment]
@younatics 

awesome! looks good!

have you thought of creating a feature update later on for: "close all expanding" and "opening all expanding" ?

**Comment by younatics** [comment]
I have :) I will do it in this weekend!

**Comment by kohdesmond** [comment]
+1

**Comment by younatics** [comment]
Hello, kohdesmond
I made expand & collapse all method I'm my project. 
You can see it in [v0.7.0](https://github.com/younatics/YNExpandableCell/releases/tag/0.7.0)
Cheers!

**Comment by kohdesmond** [comment]
@younatics awesome thanks!

btw there's a bug at line 203 of YNTableView.swift

non expanded cell are returning TRUE instead of FALSE

**Comment by kohdesmond** [comment]
should be
line 203 : delegate.tableView(self, didSelectRowAt: indexPath, isExpandedCell: false)
line 204 : didSelectRowAtSended = true

because non-expanded shouldn't call deselect as I assumed is a normal UITableView behaviour.

**Comment by younatics** [comment]
I will open this issue again

**Comment by younatics** [comment]
Maybe, I think `isExpandedCell ` method seems confusing because expanded cell and expandable cell returns same value. I purpose to make this function because `didSelect` method and `didDeselect` method should return same `isExpandedCell ` value. But I will consider adding `isExpandedableCell ` method too 👍

**Comment by kohdesmond** [comment]
@younatics no worries, i'm glad to help you test out your new git!

**Comment by younatics** [comment]
Hello @kohdesmond ,
I published [v0.8.0](https://github.com/younatics/YNExpandableCell/releases/tag/0.8.0) with new `didSelectRowAt` and `didDeselectRowAt` method :)
Cheers!
